aspnetcore/fundamentals/index.md

```diff
@@ -16,97 +16,121 @@ ms.custom: H1Hack27Feb2017
     public void Configure(IApplicationBuilder app)
     {
     }
 }
 ```
 
-* `ConfigureServices` defines the services (see [Services](#services) below) used by your app (such as the ASP.NET Core MVC framework, Entity Framework Core, Identity, etc.)
+* `ConfigureServices` defines the services (see [Services](#services) below) used by your application (such as the ASP.NET Core MVC, Entity Framework Core, Identity, etc.).
 
-* `Configure` defines the [middleware](middleware.md) in the request pipeline
+* `Configure` defines the [middleware](xref:fundamentals/middleware) in the request pipeline.
 
-For more information, see [Application startup](startup.md).
+For more information, see [Application startup](xref:fundamentals/startup).
 
 ## Services
 
-A service is a component that is intended for common consumption in an application. Services are made available through dependency injection (DI). ASP.NET Core includes a simple built-in inversion of control (IoC) container that supports constructor injection by default. The built-in container can be easily replaced with your container of choice. In addition to its loose coupling benefit, DI makes services available throughout your app. For example, [logging](logging.md) is available throughout your app.
+A service is a component that is intended for common consumption in an application. Services are made available through dependency injection (DI). ASP.NET Core includes a simple, native inversion of control (IoC) container that supports [constructor injection](xref:mvc/controllers/dependency-injection#constructor-injection) by default. The native container can be easily replaced with your container of choice. In addition to its loose coupling benefit, DI makes services available throughout your application. For example, [logging](xref:fundamentals/logging) is available throughout your application.
 
-For more information, see [Dependency injection](dependency-injection.md) .
+For more information, see [Dependency injection](xref:fundamentals/dependency-injection).
 
 ## Middleware
 
-In ASP.NET Core you compose your request pipeline using [Middleware](middleware.md). ASP.NET Core middleware performs asynchronous logic on an `HttpContext` and then either invokes the next middleware in the sequence or terminates the request directly. You generally "Use" middleware by taking a dependency on a NuGet package and invoking a corresponding `UseXYZ` extension method on the `IApplicationBuilder` in the `Configure` method.
+In ASP.NET Core, you compose your request pipeline using [Middleware](xref:fundamentals/middleware). ASP.NET Core middleware performs asynchronous logic on an `HttpContext` and then either invokes the next middleware in the sequence or terminates the request directly. You generally "Use" middleware by taking a dependency on a NuGet package and invoking a corresponding `UseXYZ` extension method on the `IApplicationBuilder` in the `Configure` method.
```

Comment:
Do we need to talk about taking another NuGet dependency? 
Maybe something like:
A middlware called `XYZ` is typically added by calling `UseXYZ`  in the config method.
>`UseXYZ` extension method on the `IApplicationBuilder`
That's steve speak. We want to avoid being a compiler and verbalizing code.